Board.rb

```diff
@@ -0,0 +1,394 @@
+
+  end
+
+  def capture_piece(location)
+    @captured << square(location[0], location[1])
+    @board[location[0]][location[1]] = nil
+  end
+
+
+  def piece_captured?(piece, location)
+    (@board[location[0]][location[1]] != nil)
+  end
+
+  def capture_piece(location)
+    @captured << square(location[0], location[1])
+    @board[location[0]][location[1]] = nil
+    @game_over = true if king_taken?
+  end
+
+  def valid_move(piece)
+    valid_moves = []
+    piece.respond_to?(:pawn_attack) ? piece_moves = piece.moves(@board) : piece_moves = piece.moves
+    current_location = piece.location
+    piece_moves.each do |move|
+      x = current_location[0] + move[0]
+      y = current_location[1] + move[1]
+      next if out_of_bounds?([x,y])
+      if free_space?(piece, x, y)
+        valid_moves << [x,y] #fix
+        vector_array = check_direction(piece, x, y, move[0], move[1])
+        vector_array.each { |coord| valid_moves << coord } unless piece.multiple_moves == false || vector_array == []
+        # valid_moves << vector_array
+      elsif (@board[x][y]).color != piece.color
+        valid_moves << [x,y]
+      else
+        next
+      end
+    end
+    valid_moves
+  end
+
+
+  # private
+
+
+  def out_of_bounds?(location)
+    x = location[0]
+    y = location[1]
+    (x < 0 || y < 0 || x > 7|| y > 7)
+  end
+
+
+  def king_taken?
+    @captured.each do |object|
+      return true if object.name == "king"
+    end
+  end
+
+
+  def check_direction(piece, x, y, add_x, add_y, array_direction = [])
+    if out_of_bounds?([x + add_x, y + add_y])
+      return array_direction
+    elsif friendly_fire?(piece, x + add_x, y + add_y) #&& piece.name != 'knight'
+      return array_direction
+    elsif free_space?(piece,x + add_x, y + add_y)
+      array_direction << [x + add_x, y + add_y]
+      check_direction(piece, x + add_x, y + add_y, add_x, add_y, array_direction )
+    else
+      array_direction << [x + add_x, y + add_y]
+    end
+    array_direction
+  end
+
+  def find_piece(location)
+    piece = square(location[0], location[1])
+  end
+
+  def check?(player, board_state=board) #does player color put the other teams king in check?
+    result = false
+    team = all_pieces_same_color(player)# all of one player's pieces on board
+    all_possible_team_moves = [] # all the potential moves by all the player's pieces
+    team.each do |piece|
+      valid_move(piece).each do |move|
+        x = move[0]
+        y = move[1]
+        all_possible_team_moves << move
+        next if board_state[x][y] == nil
+        if board_state[x][y].name == "king" #if one of your valid moves equals the king
+          king_location = [x,y] # location, the king is in check
+          result = true
+          checkmate?(player, all_possible_team_moves, king_location)
+        end
+
+      end
+    end
+    result
+  end
+
+  def checkmate?(player, all_possible_team_moves, king_location)
+    king_x = king_location[0]
```

Comment:
`king_x, king_y = king_location` This pattern could probably save a lot of typing throughout. For example

`row, col = location` instead of using `location[0]` and `location[1]` all over.